### ISSUE TITLE
AWS SSM vs AWS Secretsmanager > Strange behaviour if only ENV AWS_REGION is set

### ISSUE THREAD

**Issue by nexeck** [concourse/concourse#2191]
Hi there!

# Bug Report

I am using AWS SSM Parameter Store for Secrets. The Web Instance tries to access SSM and Secretsmanager

Steps to reproducude:
- Just set the ENV for AWS_REGION, both credential backends get activated

SSM:
https://github.com/concourse/atc/blob/master/creds/ssm/manager.go#L23
https://github.com/concourse/atc/blob/master/creds/ssm/manager.go#L49

Secretsmanager:
https://github.com/concourse/atc/blob/master/creds/secretsmanager/manager.go#L23
https://github.com/concourse/atc/blob/master/creds/secretsmanager/manager.go#L46

The following can also be handy:

* Concourse version: 3.12.0
* Deployment type (BOSH/Docker/binary): docker
* Infrastructure/IaaS: AWS
* Browser (if applicable):
* Did this used to work?

**Comment by owen-fugue** [comment]
@nexeck Thanks for doing the due diligence here. I came across your issue after I'd been debugging this as well. I think some kind of signaling value, as you suggested, would make sense.

**Comment by jama-pivotal** [comment]
@nexeck thanks for doing the research. I've scheduled it under Operations.

**Comment by sjiveson** [comment]
I've seen the same behaviour (with ATC randomly attempting to pull secrets from one or the other each time I launched it) and had to add these (set to default) values to force use of Secrets Manager, I assume the same will work for SSM until you implement a fix:

```
CONCOURSE_AWS_SECRETSMANAGER_PIPELINE_SECRET_TEMPLATE: "/concourse/{{.Team}}/{{.Pipeline}}/{{.Secret}}"
        CONCOURSE_AWS_SECRETSMANAGER_TEAM_SECRET_TEMPLATE: "/concourse/{{.Team}}/{{.Secret}}"
```

**Comment by analytically** [comment]
Seeing strange behaviour as well, the secret id is extracted from the arn? Can't seem to get it to work.

**Comment by sjiveson** [comment]
Secret ID? Can you elaborate?

**Comment by Luke-Vear** [comment]
Getting this issue on concourse 4.2.1

**Comment by vito** [comment]
Ended up running in to this and fixing it in our quest to use the binaries in the BOSH release (#2575). It'll be in the next version. Thanks for the sleuthing!

**Comment by bracki** [comment]
@vito How does #2575 solve this very issue? Shouldn't there be the ability to tell Concourse exactly which credentials backend to use?

**Comment by vito** [comment]
@bracki the offending env var (`AWS_REGION`) has been removed in favor of a more clearly named var for each (following the existing `CONCOURSE_` prefix scheme)

**Comment by Jeeppler** [comment]
@Luke-Vear do you still have this issue in Concourse 4.2.2?

**Comment by vito** [comment]
@benabineri Yes, you should avoid setting `AWS_REGION`. This was fixed in v5.0.0.

**Comment by eedwards-sk** [comment]
@vito SSM docs still say `AWS_REGION`: https://concourse-ci.org/aws-ssm-credential-manager.html

and ASM docs say `CONCOURSE_AWS_SECRETSMANAGER_REGION`: https://concourse-ci.org/aws-asm-credential-manager.html

however this issue seems to claim SSM should use `CONCOURSE_AWS_SSM_AWS_REGION`: https://github.com/concourse/concourse/issues/2003

which differs from the ASM version (`_REGION` vs `_AWS_REGION`)

Which one is correct?
